### ISSUE TITLE
v14 --fix rewrites `a 'b' c` to 'a \'b\' c' instead of "a 'b' c"

### ISSUE THREAD

**Issue by wmhilton** [standard/standard#1380]
**What version of this package are you using?**
v14.0.0

**What operating system, Node.js, and npm version?**
node v10.15.3

**What happened?**
`standard --fix` rewrote my template strings (containing single quotes) into single quote strings (containing escaped single quotes):

e.g.
```diff
-    expect(await fs.exists(`${dir}/a.txt`)).toBe(true, `'a.txt' exists`)
+    expect(await fs.exists(`${dir}/a.txt`)).toBe(true, '\'a.txt\' exists')
```

**What did you expect to happen?**
I didn't expect that. Like... I'm a little sad that it rewrote my template strings at all, but short of reverting #838, at _worst_ --fix should rewrite them using double quotes like,

```diff
+    expect(await fs.exists(`${dir}/a.txt`)).toBe(true, "'a.txt' exists")
```

**Are you willing to submit a pull request to fix this bug?**
Maybe? If someone can point me at what file does the `--fix` behavior shouldn't be too hard for me to figure out.

**Comment by feross** [comment]
I agree with you that this fix is not ideal. Double quotes would be better. The fix for this would need to go into ESLint in this file: https://github.com/eslint/eslint/blob/master/lib/rules/quotes.js It looks like we'd need to take into account the value of the `avoidEscape` rule setting whenever converting between quote types in `--fix`. This seems like a reasonable change.

@wmhilton As a next step, would you like to open this as an issue on ESLint to see what they think of the idea?

**Comment by wmhilton** [comment]
> As a next step, would you like to open this as an issue on ESLint to see what they think of the idea?

done

**Comment by feross** [comment]
Cool, let's wait to see how that issue discussion goes: https://github.com/eslint/eslint/issues/12129

**Comment by feross** [comment]
@wmhilton FYI, there's a new issue opened with a proposed solution to this and other problems with the `quotes` rule: https://github.com/eslint/eslint/issues/12156

**Comment by feross** [comment]
@wmhilton That seems like the right rule.

**Comment by FKPSC** [comment]
I just had a case of standard autofixing this:
```db.raw(`date_trunc('day', "table"."column") as day`), ```
to
```db.raw('date_trunc(\'day\', "purchaseOrderItem"."purchaseOrderItemEmptyCaptured") as day'),```


I was really surprised as this is the second rule displayed on the [website](https://standardjs.com/rules.html#rules):

> Use single quotes for strings except to avoid escaping.
> 
> eslint: [quotes](http://eslint.org/docs/rules/quotes)
> 
> ```console.log('hello there')```    // ✓ ok
> ```console.log("hello there")```    // ✗ avoid
> ```console.log(`hello there`)```    // ✗ avoid
> 
> ```$("<div class='box'>")```        // ✓ ok
> ```console.log(`hello ${name}`)```  // ✓ ok
> > 
I would at least expect not autofix (just like for `eqeqeq`) in these cases, or an update to the documented rules.

**Comment by FKPSC** [comment]
I just had a case of standard autofixing this:
```db.raw(`date_trunc('day', "table"."column") as day`), ```
to
```db.raw('date_trunc(\'day\', "table"."column") as day'),```


I was really surprised as this is the second rule displayed on the [website](https://standardjs.com/rules.html#rules):

> Use single quotes for strings except to avoid escaping.
> 
> eslint: [quotes](http://eslint.org/docs/rules/quotes)
> 
> ```console.log('hello there')```    // ✓ ok
> ```console.log("hello there")```    // ✗ avoid
> ```console.log(`hello there`)```    // ✗ avoid
> 
> ```$("<div class='box'>")```        // ✓ ok
> ```console.log(`hello ${name}`)```  // ✓ ok
> > 
I would at least expect not autofix (just like for `eqeqeq`) in these cases, or an update to the documented rules.

**Comment by karlismelderis-mckinsey** [comment]
I had same surprise.
I have plenty of "`" left behind for strings that are not templates.

So I tried to set `allowTemplateLiterals` to false and sadly code was full with escapes `\` because we do have strings with both `'` and `"`.
Final result was way more ugly so I had to leave backticks in place even for strings that don't need it any more.

Can we please make `avoidEscape` to prefer backticks over escaping string.
